osu.Game/Screens/Play/ReplayPlayer.cs

```diff
@@ -1,41 +1,42 @@
 ﻿// Copyright (c) ppy Pty Ltd <contact@ppy.sh>. Licensed under the MIT Licence.
 // See the LICENCE file in the repository root for full licence text.
 
+using System;
+using System.Collections.Generic;
 using System.Threading.Tasks;
 using osu.Framework.Input.Bindings;
+using osu.Game.Beatmaps;
 using osu.Game.Input.Bindings;
+using osu.Game.Rulesets.Mods;
 using osu.Game.Scoring;
 using osu.Game.Screens.Ranking;
 
 namespace osu.Game.Screens.Play
 {
     public class ReplayPlayer : Player, IKeyBindingHandler<GlobalAction>
     {
-        protected readonly Score Score;
+        private readonly Func<IBeatmap, IReadOnlyList<Mod>, Score> createScore;
 
         // Disallow replays from failing. (see https://github.com/ppy/osu/issues/6108)
         protected override bool CheckModsAllowFailure() => false;
 
         public ReplayPlayer(Score score, PlayerConfiguration configuration = null)
+            : this((_, __) => score, configuration)
+        {
+        }
+
+        public ReplayPlayer(Func<IBeatmap, IReadOnlyList<Mod>, Score> createScore, PlayerConfiguration configuration = null)
             : base(configuration)
         {
-            Score = score;
+            this.createScore = createScore;
         }
 
         protected override void PrepareReplay()
         {
             DrawableRuleset?.SetReplayScore(Score);
         }
```

Comment:
I meant... This entire override doesn't need to exist anymore, does it?

Reply:
Ah. So you're saying we should allow the re-recording of frames to the score target..? I'm not sure if that will work but can try if that's your intention here.

Reply:
OH it's a completely different method!
```
SetReplayScore()
SetRecordTarget()
```
>_<

Reply:
I'm wondering if the naming should be changed, or if the method is combined with a `bool shouldRecord`. Not sure..